### PR TITLE
[6.1.x] tele build

### PR DESCRIPTION
**Pull request by a-palchikov**
## Description
This PR fixes a regression in custom planet handling which filtered planet package from the list of package dependencies and did not handle the case of the enterprise hub that returns application manifests stripped of any new configuration including the new planet package configuration in `systemOptions`.

## Type of change

* Bug fix (non-breaking change which fixes an issue)
* Regression fix (non-breaking change which fixes a regression)
* This change has a user-facing impact

## Linked tickets and other PRs

* Closes https://github.com/gravitational/gravity/issues/2125

## TODOs

- [x] Self-review the change
- [ ] Write tests
- [x] Perform manual testing
- [ ] Address review feedback

## Testing done

TODO. I'm working on a set of tests to avoid breaking this behavior in future.

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

File: lib/builder/builder_test.go [hunk 1]
```diff
@@ -97,6 +112,312 @@ func (s *BuilderSuite) TestSelectRuntimeVersion(c *check.C) {
...
+            count: 2
+nodeProfiles:
+  - name: master
+    labels:
+      node-role.kubernetes.io/master: "true"
+  - name: node
+    labels:
+      node-role.kubernetes.io/node: "true"
+systemOptions:
+  #dependencies:
```
Comment (r0mant) [inline lib/builder/builder_test.go:213]: Remove?

**Review by r0mant — APPROVED** [review]

File: lib/builder/build.go [hunk 2]
```diff
@@ -74,6 +74,7 @@ func Build(ctx context.Context, builder *Builder) error {
 		err = builder.SyncPackageCache(ctx, runtimeVersion)
 		if err != nil {
 			if trace.IsNotFound(err) {
+				logrus.WithError(err).Warn("Failed to sync package cache.")
```
Comment (knisbet) [inline lib/builder/build.go:77]: Include runtimeVersion in the log?

**Review by knisbet — APPROVED** [review]

File: lib/builder/builder_test.go (hunk 1 above)
Comment (a-palchikov) [reply lib/builder/builder_test.go:213]: I've added a couple of tests where this is used, so I'll update the branch shortly.

File: lib/builder/build.go (hunk 2 above)
Comment (a-palchikov) [reply lib/builder/build.go]: Done in 51bec856

**Comment by a-palchikov** [comment]
Discovered a couple of data races including one in go-dockerclient when I added the additional tests for custom planet package handling for `tele build` so I bumped the package as well.

File: lib/builder/builder_test.go [hunk 3]
```diff
@@ -97,6 +112,312 @@ func (s *BuilderSuite) TestSelectRuntimeVersion(c *check.C) {
...
 
+func (s *BuilderSuite) TestBuildInstallerWithDefaultPlanetPackage(c *check.C) {
+	// setup
+	remoteEnv := newEnviron(c)
+	defer remoteEnv.Close()
+	stateDir := c.MkDir()
+	appDir := c.MkDir()
+	manifestPath := filepath.Join(appDir, defaults.ManifestFileName)
+	manifestBytes := []byte(`
+apiVersion: bundle.gravitational.io/v2
```
[the remark this answers is not in the document]
Comment (a-palchikov) [reply lib/builder/builder_test.go:140]: Sure, let me tweak these.

**Pull request merged**